src/Bongo/Actors/Query/QueryWorker.cs

```diff
@@ -37,9 +38,21 @@ private void ProcessQueryRequest(QueryRequest request, IActorRef sender)
             {
                 try
                 {
-                    var response = await lease.Connection.Ask<QueryResponse>(new LeaseQuery(request.QueryString, lease.ConnectionLeaseId));
-                    var parsedResponse = ParseResults(response, request.ResponseType);
-                    sender.Tell(parsedResponse);
+                    var response =
+                        await lease.Connection.Ask(new LeaseQuery(request.QueryString, lease.ConnectionLeaseId));
+                    switch (response)
+                    {
+                        case QueryResponse queryResponse:
+                            var parsedResponse = ParseResults(queryResponse, request.ResponseType);
+                            sender.Tell(parsedResponse);
+                            break;
+                        case BeeswaxException exception:
+                            sender.Tell(exception);
+                            break;
+                        case Exception exception:
```

Comment:
Wouldn't this case cover BeeswaxException also?

Reply:
Pretty sure it would. I may want to break out unexpected exceptions in the future